src/Api/Vault/AuthorizationHandlers/Collections/BulkCollectionAuthorizationHandler.cs

```diff
@@ -225,25 +225,41 @@ private async Task<bool> CanUpdateGroupAccessAsync(ICollection<Collection> resou
 
     private async Task<bool> CanDeleteAsync(ICollection<Collection> resources, CurrentContextOrganization? org)
     {
-        // Owners, Admins, and users with DeleteAnyCollection permission can always delete collections
-        if (org is
-        { Type: OrganizationUserType.Owner or OrganizationUserType.Admin } or
-        { Permissions.DeleteAnyCollection: true })
+        // Users with DeleteAnyCollection permission can always delete collections
+        if (org is { Permissions.DeleteAnyCollection: true })
         {
             return true;
         }
 
-        // Check for non-null org here: the user must be apart of the organization for this setting to take affect
-        // The limit collection management setting is disabled,
-        // ensure acting user has manage permissions for all collections being deleted
-        if (await GetOrganizationAbilityAsync(org) is { LimitCollectionCreationDeletion: false })
+        var organizationAbility = await GetOrganizationAbilityAsync(org);
+
+        // If AllowAdminAccessToAllCollectionItems is true, Owners and Admins can delete any collection, regardless of LimitCollectionCreationDeletion setting
+        if ((organizationAbility is { AllowAdminAccessToAllCollectionItems: true } ||
+             !_featureService.IsEnabled(FeatureFlagKeys.FlexibleCollectionsV1)) &&
+            org is { Type: OrganizationUserType.Owner or OrganizationUserType.Admin })
```

Comment:
I ended up separating these lines to make the logic a little clearer - see [CanUpdateCollectionAsync](https://github.com/bitwarden/server/blob/6af746c6263c0a3feaa8e39a85c5a05bce2d8afb/src/Api/Vault/AuthorizationHandlers/Collections/BulkCollectionAuthorizationHandler.cs#L193-L196).